### PR TITLE
enhance: use `flag-icons` for country flags

### PR DESCRIPTION
**Pull request by marcelgerber**
Updating our flags with 4x3 variants taken from https://flagicons.lipis.dev/.
They are mostly pretty nice-looking and up-to-date, although there are a few exceptions.

Pulling these files into our own repo like this allows us to easily replace a flag on our own if we ever need to do so (although we should try our best to create a 4x3 variant ourselves).

### CODE REVIEW

File: public/images/flags/BLM.svg [hunk 1]
```diff
@@ -1,160 +1,5 @@
...
-<circle fill="#CCC" stroke="#9E7800" stroke-width="0.5" cx="458.079" cy="447.518" r="1.668"/>
-<circle fill="#FFF" cx="457.793" cy="447.026" r="0.677"/>
-<circle fill="#CCC" stroke="#9E7800" stroke-width="0.5" cx="481.16" cy="442.665" r="1.668"/>
-<circle fill="#FFF" cx="480.874" cy="442.174" r="0.677"/>
-<circle fill="#CCC" stroke="#9E7800" stroke-width="0.5" cx="481.16" cy="447.518" r="1.668"/>
-<circle fill="#FFF" cx="480.874" cy="447.026" r="0.677"/>
-</g>
-<use xlink:href="#t3" x="-49.151" y="-55.104"/>
-<use xlink:href="#t3" x="49.151" y="-55.104"/>
+<svg xmlns="http://www.w3.org/2000/svg" id="flag-icons-bl" viewBox="0 0 640 480">
```
Comment (ikesau) [inline public/images/flags/BLM.svg:1]: How come this one's been changed to 🇫🇷 but Guadeloupe's been deleted?

File: public/images/flags/KGZ.svg [hunk 2]
```diff
@@ -1,29 +1,15 @@
...
-<circle r="27" fill="#e8112d"/>
-<circle cy="-1.5" r="22.5" fill="#ffef00"/>
-<g id="x" fill="#e8112d" clip-path="url(#c)">
-<path d="M-23.055236,18a37.755236,37.755236 0 1,1 75.510472,0H52.760449a36.260449,36.260449 0 1,0 -72.520898,0z"/>
-<path d="M-17.601203,18a35.301203,35.301203 0 1,1 70.602406,0H53.424578a33.924578,33.924578 0 1,0 -67.849156,0z"/>
-<path d="M20.7,-15.052356a33.052356,33.052356 0 1,0 0,66.104712L22.5,49.819805a31.819805,31.819805 0 1,1 0,-63.63961z"/>
-</g>
-<use xlink:href="#x" transform="scale(-1,1)"/>
-</g>
+<svg xmlns="http://www.w3.org/2000/svg" id="flag-icons-kg" viewBox="0 0 640 480">
```
Comment (ikesau) [inline public/images/flags/KGZ.svg:1]: Another one to try and update upstream

File: public/images/flags/KIR.svg [hunk 3]
```diff
@@ -1,21 +1,36 @@
...
-<path fill="none" d="M286.53,40.436c-1.63-8.511-9.6-11.065-19.66-12.594-10.95-1.663-22.12-2.589-33.04-5.607-2.71-0.748-5.23-2.07-7.85-3.105"/>
-<path fill="none" stroke-linecap="round" d="M290.02,52.983c10.28,0.754,17.91-2.336,26.68-6.706"/>
-<path fill="none" stroke-linecap="round" d="M254.63,41.831h12.96"/>
-<circle fill="#bd9c08" stroke="none" cx="271.75" cy="40.437" r="1.62"/>
-</g>
-<rect fill="#003f87" width="600" height="135" y="165"/>
-<path id="w" fill="#fff" d="M0,257.14c20,0,40,12.86,60,12.86s40-12.86,60-12.86,40,12.86,60,12.86,40-12.86,60-12.86,40,12.86,60,12.86,40-12.86,60-12.86,40,12.86,60,12.86,40-12.86,60-12.86,40,12.86,60,12.86,40-12.86,60-12.86v-21.43c-20,0-40,12.86-60,12.86s-40-12.86-60-12.86-40,12.86-60,12.86-40-12.86-60-12.86-40,12.86-60,12.86-40-12.86-60-12.86-40,12.86-60,12.86-40-12.86-60-12.86-40,12.86-60,12.86-40-12.86-60-12.86v21.43z"/>
-<use xlink:href="#w" y="-42.8571"/>
-<use xlink:href="#w" y="-85.7142"/>
+<svg xmlns="http://www.w3.org/2000/svg" id="flag-icons-ki" viewBox="0 0 640 480">
```
Comment (ikesau) [inline public/images/flags/KIR.svg:1]: 🥴

File: public/images/flags/KWT.svg [hunk 4]
```diff
@@ -1,6 +1,13 @@
-<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 6">
-<rect width="12" height="2" fill="#007a3d"/>
-<rect width="12" height="2" y="2" fill="#fff"/>
-<rect width="12" height="2" y="4" fill="#ce1126"/>
-<polygon points="0,0 3,2 3,4 0,6"/>
+<svg xmlns="http://www.w3.org/2000/svg" id="flag-icons-kw" viewBox="0 0 640 480">
```
Comment (ikesau) [inline public/images/flags/KWT.svg:1]: Another weird colour

File: public/images/flags/ERI.svg [hunk 5]
```diff
@@ -1,6 +1,8 @@
...
+<svg xmlns="http://www.w3.org/2000/svg" id="flag-icons-er" viewBox="0 0 640 480">
```
Comment (ikesau) [inline public/images/flags/ERI.svg:1]: Flagging (ha) all the weird colours I spot so we can maybe do a batch update in an upstream PR

File: public/images/flags/GIN.svg [hunk 6]
```diff
@@ -1,5 +1,7 @@
-<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 3 2">
-<rect width="1" height="2" fill="#ce1126"/>
-<rect width="1" height="2" x="1" fill="#fcd116"/>
-<rect width="1" height="2" x="2" fill="#009460"/>
+<svg xmlns="http://www.w3.org/2000/svg" id="flag-icons-gn" viewBox="0 0 640 480">
+  <g fill-rule="evenodd" stroke-width="1pt">
+    <path fill="red" d="M0 0h213.3v480H0z"/>
+    <path fill="#ff0" d="M213.3 0h213.4v480H213.3z"/>
+    <path fill="#090" d="M426.7 0H640v480H426.7z"/>
```
Comment (ikesau) [inline public/images/flags/GIN.svg:5]: Seems too vivid

File: public/images/flags/MDA.svg [hunk 7]
```diff
@@ -1,140 +1,70 @@
...
-<use xlink:href="#leaf" transform="rotate(72,838.89,486.468)"/>
-<use xlink:href="#leaf" transform="rotate(144,838.89,486.468)"/>
-<use xlink:href="#leaf" transform="rotate(216,838.89,486.468)"/>
-<use xlink:href="#leaf" transform="rotate(288,838.89,486.468)"/>
-<circle cx="838.890" cy="486.468" r="7.065"/>
-<path d="m962.925,476.942c-11.729,2.268-20.17,12.628-18.942,24.727-3.738-3.096-6.473-7.266-6.962-11.877-0.951-8.979,5.571-15.922,15.4095-15.922,3.7615,0,7.3755,1.152,10.4945,3.072z"/>
-</g>
-</g>
-</g>
+<svg xmlns="http://www.w3.org/2000/svg" id="flag-icons-md" viewBox="0 0 640 480">
```
Comment (ikesau) [inline public/images/flags/MDA.svg:1]: 🤠

File: public/images/flags/MHL.svg [hunk 8]
```diff
@@ -1,19 +1,7 @@
...
-<use xlink:href="#r1" transform="rotate(15)"/>
-<use xlink:href="#r1" transform="rotate(30)"/>
-<use xlink:href="#r1" transform="rotate(45)"/>
-<use xlink:href="#r1" transform="rotate(60)"/>
-</g>
-<use xlink:href="#r6" transform="rotate(90)"/>
-<use xlink:href="#r6" transform="rotate(180)"/>
-<use xlink:href="#r6" transform="rotate(270)"/>
-</g>
+<svg xmlns="http://www.w3.org/2000/svg" id="flag-icons-mh" viewBox="0 0 640 480">
```
Comment (ikesau) [inline public/images/flags/MHL.svg:1]: 🇲🇭

File: public/images/flags/MLT.svg [hunk 9]
```diff
@@ -1,49 +1,58 @@
...
-<path d="m116.91,117.16s-0.4868,0.3587-1.1273,0c-0.33307-0.76862-1.2298-0.81987-1.2298-0.81987s-0.57758-1.6593-1.486-1.1786c-0.14091-0.8583-1.1273-0.40993-1.1273-0.40993s-0.0256-1.2426-1.1786-1.0761c0.38431-1.422-1.5885-1.3323-1.5885-1.3323s-0.064-1.6013-2.972-1.281c6.2643-2.8055,11.222,4.4068,11.222,4.4068"/>
-<path d="m119.88,117.77s1.4476,1.6269,2.2034,1.6397c0.75582,0.0128,0.81987-1.5885,0.81987-1.5885s0.2434,0.69176,0.6149,0.51242c-0.0384,0.48679,0.46118,0.6149,0.46118,0.6149s-1.3067,3.0361-5.0217,0.15372"/>
-<path d="m108.55,118.39-0.51242,2.1009s-0.65333,0.58928-0.46118,0.71739c0.19216,0.1281,1.2298-0.20497,1.2298-0.20497s1.6141-0.19215,2.1009-0.66614c0.33308,0.75582,1.0761,0.81987,1.0761,0.81987s0.0512-0.32026-0.0512-0.51242c0.35869,0.0641,0.66615-0.0512,0.66615-0.0512s0.20496-0.58929-0.25621-0.81987c0.0512-0.43556,0.26901-0.81987-0.51242-0.76863-0.57647,0.6149-1.9984,1.3195-2.3571,1.0761-0.35869-0.2434,1.4092-1.0248,0.92236-2.0497"/>
-</g>
-</g>
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" id="flag-icons-mt" viewBox="0 0 640 480">
```
Comment (ikesau) [inline public/images/flags/MLT.svg:1]: This crest seems wrong

File: public/images/flags/MOZ.svg [hunk 10]
```diff
@@ -1,13 +1,21 @@
...
+<svg xmlns="http://www.w3.org/2000/svg" id="flag-icons-mz" viewBox="0 0 640 480">
```
Comment (ikesau) [inline public/images/flags/MOZ.svg:1]: Colours

File: public/images/flags/MLI.svg [hunk 11]
```diff
@@ -1,5 +1,7 @@
-<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 450 300">
-<rect fill="#CE1126" width="450" height="300"/>
-<rect fill="#FCD116" width="300" height="300"/>
-<rect fill="#14B53A" width="150" height="300"/>
+<svg xmlns="http://www.w3.org/2000/svg" id="flag-icons-ml" viewBox="0 0 640 480">
+  <g fill-rule="evenodd">
+    <path fill="red" d="M425.8 0H640v480H425.7z"/>
+    <path fill="#009a00" d="M0 0h212.9v480H0z"/>
+    <path fill="#ff0" d="M212.9 0h214v480h-214z"/>
```
Comment (ikesau) [inline public/images/flags/MLI.svg:5]: Colours

File: public/images/flags/NAM.svg [hunk 12]
```diff
@@ -1,17 +1,16 @@
...
-<path id="r" d="M0,10 1.552914,5.795555H-1.552914z"/>
-<use xlink:href="#r" transform="rotate(90)"/>
-<use xlink:href="#r" transform="rotate(180)"/>
-<use xlink:href="#r" transform="rotate(270)"/>
-</g>
-<use xlink:href="#r4" transform="rotate(30)"/>
-<use xlink:href="#r4" transform="rotate(60)"/>
-<circle r="5.5" stroke="#003580"/>
-</g>
+<svg xmlns="http://www.w3.org/2000/svg" id="flag-icons-na" viewBox="0 0 640 480">
```
Comment (ikesau) [inline public/images/flags/NAM.svg:1]: Colours

File: public/images/flags/SVN.svg [hunk 13]
```diff
@@ -1,22 +1,18 @@
...
-<use xlink:href="#wave" transform="translate(0 17.321)"/>
-<g id="s" transform="translate(0,-120) scale(2.25)">
-<path stroke-width=".2" d="m0-5l1 3.2679 3.3301-0.7679-2.3301 2.5 2.3301 2.5-3.3301-0.7679-1 3.2679-1-3.2679-3.3301 0.7679 2.3301-2.5-2.3301-2.5 3.3301 0.7679z" fill="#fd0"/>
-</g>
-<use xlink:href="#s" transform="translate(-33.75,-45)"/>
-<use xlink:href="#s" transform="translate(33.75,-45)"/>
-<path d="m-111.58-167.05l9.96 146.99a146.95 146.95 0 0 0 101.62 129.95 146.95 146.95 0 0 0 101.62 -129.95l9.96-146.99a280.22 280.22 0 0 0 8.42 3.82l-9.74 143.75a155.61 155.61 0 0 1 -110.26 138.45 155.61 155.61 0 0 1 -110.26 -138.45l-9.74-143.75a280.22 280.22 0 0 0 8.42 -3.82" fill="#ed1c24"/>
-</svg>
-</g>
+<svg xmlns="http://www.w3.org/2000/svg" id="flag-icons-si" viewBox="0 0 640 480">
```
Comment (ikesau) [inline public/images/flags/SVN.svg:1]: Colours

File: public/images/flags/TGO.svg [hunk 14]
```diff
@@ -1,16 +1,14 @@
...
-<g id="c">
-<path id="t" d="M 0,-95 V 0 H 50" transform="rotate(18 0,-95)"/>
-<use xlink:href="#t" transform="scale(-1,1)"/>
-</g>
-<use xlink:href="#c" transform="rotate(72)"/>
-<use xlink:href="#c" transform="rotate(144)"/>
-<use xlink:href="#c" transform="rotate(216)"/>
-<use xlink:href="#c" transform="rotate(288)"/>
-</g>
+<svg xmlns="http://www.w3.org/2000/svg" id="flag-icons-tg" viewBox="0 0 640 480">
```
Comment (ikesau) [inline public/images/flags/TGO.svg:1]: Colours

File: public/images/flags/UKR.svg [hunk 15]
```diff
@@ -1,4 +1,6 @@
-<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1200 800">
-<rect width="1200" height="800" fill="#005BBB"/>
-<rect width="1200" height="400" y="400" fill="#FFD500"/>
+<svg xmlns="http://www.w3.org/2000/svg" id="flag-icons-ua" viewBox="0 0 640 480">
+  <g fill-rule="evenodd" stroke-width="1pt">
+    <path fill="gold" d="M0 0h640v480H0z"/>
```
Comment (ikesau) [inline public/images/flags/UKR.svg:3]: Colours

File: public/images/flags/VCT.svg [hunk 16]
```diff
@@ -1,10 +1,8 @@
-<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 72 48">
-<rect width="72" height="48" fill="#009e60"/>
-<rect width="54" height="48" fill="#fcd116"/>
-<rect width="18" height="48" fill="#0072c6"/>
-<g fill="#009e60">
-<path d="m32,34 4,8 4,-8 -4,-8z"/>
-<path d="m27,24 4,8 4,-8 -4,-8z"/>
-<path d="m37,24 4,8 4,-8 -4,-8z"/>
-</g>
+<svg xmlns="http://www.w3.org/2000/svg" id="flag-icons-vc" viewBox="0 0 640 480">
```
Comment (ikesau) [inline public/images/flags/VCT.svg:1]: Colours

**Review by ikesau — COMMENTED** [review]
Nice! I definitely want to get these in 🙂 

I wonder though, should we ensure that every country in regions.json has an associated flag? That would be a nice promise to keep for the sake of the site.

I went through and marked all the flags that seemed egregiously different from wikipedia. I feel like we should update them here (mostly with correct colours) and then submit a PR upstream. Happy to help with this if you agree 🙂

**Comment by marcelgerber** [comment]
> I wonder though, should we ensure that every country in regions.json has an associated flag? That would be a nice promise to keep for the sake of the site.

Ah yes, good point. I didn't realise that we excluded unlisted countries - now they're included, and we have flags for all of them.

File: public/images/flags/BFA.svg
Comment (marcelgerber) [inline public/images/flags/BFA.svg:1]: The star seems a bit overly large here?

File: public/images/flags/BHS.svg
Comment (marcelgerber) [inline public/images/flags/BHS.svg:1]: Colors are off

File: public/images/flags/BWA.svg
Comment (marcelgerber) [inline public/images/flags/BWA.svg:1]: Colors seem to be off

File: public/images/flags/CAF.svg
Comment (marcelgerber) [inline public/images/flags/CAF.svg:1]: Colors

File: public/images/flags/DMA.svg
Comment (marcelgerber) [inline public/images/flags/DMA.svg:1]: This is an old flag design, it seems - The yellow star outlines are no longer used.

**Review by ikesau — APPROVED** [review]
Now that I've manually overwritten 18 of these, we won't be able to run the script again without a small headache, or getting the changes merged upstream. I think that's okay though. We probably wouldn't have been running it often, anyway.

**Pull request merged**